src/locale/japanese.cpp

```diff
@@ -40,57 +40,41 @@ static const convert_key s2j_table[] = { { "mb", "nb" }, { "mp", "np" }, { "mv",
 
 /*!
  * @brief シンダリンを日本語の読みに変換する
- * @param kana 変換後の日本語文字列ポインタ
- * @param sindarin 変換前のシンダリン文字列ポインタ
+ * @param sindarin 変換前のシンダリン文字列
+ * @return std::string 変換後のシンダリン文字列
  * @details
  */
-void sindarin_to_kana(char *kana, concptr sindarin)
+std::string sindarin_to_kana(std::string_view sindarin)
 {
-    char buf[256];
-    int idx;
+    std::string kana;
 
-    sprintf(kana, "%s$", sindarin);
-    for (idx = 0; kana[idx]; idx++) {
-        if (isupper(kana[idx])) {
-            kana[idx] = (char)tolower(kana[idx]);
-        }
+    for (const auto &ch : sindarin) {
+        kana.push_back(isupper(ch) ? static_cast<char>(tolower(ch)) : ch);
     }
+    kana.append("$");
 
-    for (idx = 0; s2j_table[idx].key1 != nullptr; idx++) {
+    for (int idx = 0; s2j_table[idx].key1 != nullptr; idx++) {
```

Comment:
'auto' is desirable.
I have no comment more.

Reply:
The latest forced push changes that to use 'auto'.